code/modules/mob/living/carbon/superior_animal/roach/types/support.dm

```diff
@@ -39,6 +39,9 @@
 	if(stat != AI_inactive)
 		return
 
+	if(!target_mob)
+		return
+
 	gas_sac.add_reagent("blattedin", 1)
 
 	if(prob(7))
```

Comment:
Rather than the target mob check you have made above, have it here as `
if(target_mob && prob(7))`, otherwise you're skipping the blattedin regeneration.